### DISCUSSION TITLE
Add path to class file to circumvent cross-layer inherit

### DISCUSSION THREAD

**Discussion by jo-so** [priv-kweihmann/oelint-adv#614] [Q&A]
I have a class in a base layer that is used by bb files in other layers. Is it somehow possible to tell oelint-adv about this class like calling:

```
oelint-adv base-layer/classes/mybase.bbclass other-layer/recipe.bb
```

I've tried, but it didn't work (with v6.0). Have I done something wrong? How can I achieve this?

**Comment by priv-kweihmann** [comment]
That breaks with the idea of a self contained layer, hence requiring more knowledge of the overall build.
Requiring to almost setup a full bitbake build.
To me that would be out of scope for the linter.

It would force the need to know about layer inclusion order and stuff like that.

So short answer: by design not possible (yet).

You can ofc ship a copy of the class in question with your layer to be checked, but I think that will contradict the purpose of shared class code

**Comment by jo-so** [comment]
I came across this issue again and would like to ask, if it's possible to give oelint a list of directories it should virtually merge with the layer of the recipe. Something like `oelint-adv --more-layer meta-bsp,meta-base meta-special/recipe.bb`. It recipe.bb uses a file or distro or machine not found in meta-special, it should look in meta-base and then in meta-bsp. So it's up to the user to provide the required directories and to provide them in the right order.

**Reply by priv-kweihmann** [reply]
What is exactly the problem you are trying to solve?
Correct variable names? If so that can be done with the constants on a per layer basis.

**Reply by jo-so** [reply]
The main problem are classes and includes. In so far the layer is not self-contained, but uses classes from a base layer. We work around this with `dynamic-layers`, but this makes it more silly, because some recipes (e.g. for images) are not in the real layer but in the base layer in d-l.

Oelint-adv should not reimplement Bitbake. But I think a PATH-like thing to virtually merge in other directories in the real layer directory would be enough; like `cp -r meta-base/* meta-layer` before the start.

**Reply by priv-kweihmann** [reply]
And a PATH like exactly would replicate what bitbake does - including ordering issues and the need to calculate a layer priority.

If a layer is mandatory it should be set as part of layer dependencies and it would be safe to assume that certain classes are available

Then the constants for the layer, including information what images are (class wise) and variable names, can be put into a layer specific file.
For includes/requires it ofc will not work but IMO highlights a design problem which needs to either marked as an accepted flaw using nooelint statements or changed to move to a more self confined layer by e.g. moving to bbclasses.

From my point of view that should cover 99.9% easily - for the rest I would need a specific example.

And if a PATH like approach is the only way (what I don't believe) the libmode of the tool should provide everything that is needed to set that up
